vignettes/shared_cache.Rmd

```diff
@@ -8,45 +12,154 @@ vignette: >
+
+Workflow dependencies are connected via the <ind>**indicator files**</ind>. Recipes (e.g. R scripts) used to create the <ind>**indicator files**</ind> pull down the <data>**data files**</data> from the shared cache using `scipiper` functions such as `gd_get()`. 
+
+<build>**Build files**</build> tell remake if the <ind>**indicator files**</ind> (and other <target>**targets**</target> that are <data>**data files**</data>) and their dependencies are up-to-date. <target>**Targets**</target> that are not saved as <ind>**indicator**</ind> or <data>**data**</data> files (e.g. R objects) will not have a <build>**build status file**</build> and will be excuted every time it is called.  
+
+Example of `.ind` file dependency where the function `select_sites()` uses `gd_get()` and the <ind>**indicator file**</ind> `1_data/out/compiled_data.rds.ind` as a dependency to pull down the <data>**data file**</data> (`compiled_data.rds`) from the shared cache. 
+<pre>
+target_default: <target>1_data</target>
+
+sources: 
+  - 1_data/src/gather_data.R
+  - 1_data/src/select_sites.R
+
+targets:
+
+  <target>1_data:</target>
+     depends:
+      - <ind>1_data/out/compiled_data.rds.ind</ind>
+      - <ind>1_data/out/selected_sites.rds.ind</ind>
+
+  <ind>1_data/out/compiled_data.rds.ind:</ind>
+    command: gather_stream_data(
+      ind_file = target_name,
+      state = I("WI"), 
+      gd_config = <data>'lib/cfg/gd_config.yml'</data>)
+
+  <ind>1_data/out/selected_sites.rds.ind:</ind>
+    command: select_sites(
+      ind_file = target_name,
+      input_ind_file = <ind>'1_data/out/compiled_data.rds.ind'</ind>,
+      gd_config = <data>'lib/cfg/gd_config.yml'</data>)
+</pre>
 
-Disadvantages of a shared cache (as currently implemented):
-* In a fast-paced collaborative development environment (e.g., a 'sprint'), it is challenging to maintain synchrony between the shared cache (the data) and the git repository (the metadata). Asynchrony is not a deal-breaker but does lead to more rebuilding than would be required for a slower-paced project.
-* Though we've done much to ensure this doesn't happen, it's conceivable that metadata will become corrupt relative to the data. Some monitoring and very occasional full rebuilding is recommended when practical.
-* Old files no longer referenced by the code can accumulate on the shared cache unless manually deleted. Though these will not interfere with ongoing analysis, they can take up storage space unnecessarily.
 
 ### Guidelines
 
-Projects using a shared cache should follow these guidelines:
+####Projects using a shared cache should follow these guidelines:
+
+* Use <ind>**indicator files**</ind> (usually an `.ind` suffix) to represent most or all of the chain of connected <target>**targets**</target> within your remake files. Each `.ind` file should be one of two products of a recipe (e.g. R script), where the other product is the creation of a <data>**data file**</data>, either locally and/or in the shared cache.
```

Comment:
I'd define a recipe as an R function call rather than an R script. We sometimes write multiple functions in the same script file, and sometimes multiple recipes use the same function but each call it differently.

Reply:
Consider pointing out that the .ind file is the only product of the recipe that is declared to remake; the data file is hidden from / not announced to remake, and the purpose of the .ind file is to indirectly represent the data file to remake (where the indirectness is what enables compatibility between remake and the shared cache).

Reply:
Thanks for the clarification on that, I'll make that point.